core/src/main/java/org/mule/execution/MessageProcessorNotificationExecutionInterceptor.java

```diff
@@ -120,7 +120,7 @@ public void processExceptionReplyTo(MessagingException exception, Object replyTo
         }
         finally
         {
-            if (!NonBlockingVoidMuleEvent.getInstance().equals(result) && fireNotification)
+            if (!(result instanceof VoidMuleEvent) && fireNotification)
```

Comment:
This now matches sub-clasess of VoidMuleEvent (such as NonBlockingVoidEvent) which it didn't before.  This wasn't required before as endpoints didn't support non-blocking.  The other updates like this are for the same reason.